### ISSUE TITLE
Add slash command autocomplete functionality

### ISSUE THREAD

**Issue by iomodo** [mattermost/mattermost-plugin-github#290]
From 5.24 MM server has a slash command autocomplete functionality feature. An API can be found [here](https://github.com/mattermost/mattermost-server/blob/master/model/command_autocomplete.go#L28)
To implement the feature add `AutocompleteData` struct to the `Command` when registering a slash command.

Similar PRs:
For Todo plugin https://github.com/mattermost/mattermost-plugin-todo/pull/69
For Agenda plugin https://github.com/mattermost/mattermost-plugin-agenda/pull/40

**Comment by NateWilliams2** [comment]
Hey I'm a first-time contributor and would love to take this issue!

**Comment by hanzei** [comment]
Awesome, thanks for taking a shot @NateWilliams2 :+1: And welcome to the Mattermost Community. Please let me know, if you have any questions or need help.

**Comment by NateWilliams2** [comment]
@hanzei thanks for the welcome :) 
I have a question- I'm just trying to set up my local mattermost server with the github plugin to test it, and I'm seeing I need to enable a public site url in order to authorize a Github OAuth client to use the plugin. 
Without this set up, I'm getting `Command with a trigger of '/github' not found`. Am I right in thinking I'll need to register a domain name just to test this plugin, or do you know of a way around this?

**Comment by hanzei** [comment]
@NateWilliams2 indeed, you need a public domain for testing. I've been using https://ngrok.com/ for this. You might also want to give it a try.

**Comment by NateWilliams2** [comment]
Awesome. Thanks!
